### PR TITLE
add rtebev infer

### PR DESCRIPTION
**Pull request by wangna11BD**
add rtebev infer

### CODE REVIEW

File: docs/models/rtebev/README.md [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+ln -s /path/to/nuscenes_dataset_root ./data
+mv ./data/nuscenes_dataset_root ./data/nuscenes
+```
+
+为加速训练过程中Nuscenes数据集的加载和解析，需要事先将Nuscenes数据集里的标注信息存储在`pkl`后缀文件中。点击[下载](https://paddle3d.bj.bcebos.com/datasets/nuScenes/bevdetv2_mmdet3d.zip)PKL文件，解压到data/nuscenes目录下。
+
+下载解压后的数据集目录：
+
+```
+nuscenes_dataset_root
```
Comment (LielinJiang) [inline docs/models/rtebev/README.md]: use annotation same as bevdet

File: docs/models/rtebev/README.md [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+```
+python -m paddle.distributed.launch --gpus 0,1,2,3,4,5,6,7 tools/train.py --config configs/rtebev/rtebev_r50_nuscenes_256x704_msdepth_hybird_1f_cgbs.yml --save_dir ./output_rtebev --num_workers 4 --save_interval 1 --model ./resnet50-0676ba61.pdparams
+```
+
+训练启动参数介绍可参考文档[全流程速览](../../quickstart.md#模型训练)。
+
+#### 评估
+
+```
+python tools/evaluate.py --config configs/rtebev/rtebev_r50_nuscenes_256x704_msdepth_hybird_1f_cgbs.yml --model ./output_rtebev/epoch_20/model.pdparams --num_workers 4
```
Comment (LielinJiang) [inline docs/models/rtebev/README.md]: 这个需要--batch_size=1

File: docs/models/rtebev/README.md [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+收集shape
+```
+python infer_mf_paddletrt.py --config /path/to/config.yml --model /path/to/model.pdparams --model_file /path/to/rtebev.pdmodel --params_file /path/to/rtebev.pdiparams --use_trt --collect_shape_info
+```
+trt-fp16推理
+```
+python infer_mf_paddletrt.py --config /path/to/config.yml --model /path/to/model.pdparams --model_file /path/to/rtebev.pdmodel --params_file /path/to/rtebev.pdiparams --use_trt --trt_precision 1
+```
+
+### 使用onnx 推理nuscenes eval
```
Comment (LielinJiang) [inline docs/models/rtebev/README.md]: onnx->onnx-trt

File: paddle3d/models/detection/rtebev/rtebev.py [hunk 4]
```diff
@@ -467,19 +478,20 @@ def forward(self, samples, *args, **kwargs):
...
-    def forward_test(self,
-                    samples,
-                    #  points=None,
-                    #  img_metas=None,
-                    #  img_inputs=None,
-                     **kwargs):
+    def forward_test(
+            self,
+            samples,
+            #  points=None,
```
Comment (LielinJiang) [inline paddle3d/models/detection/rtebev/rtebev.py]: remove

File: paddle3d/models/detection/rtebev/rtebev.py [hunk 5]
```diff
@@ -534,7 +543,9 @@ def _parse_results_to_sample(self, results: dict, sample: dict):
         num_samples = len(results)
         new_results = []
         for i in range(num_samples):
-            data = Sample(None, sample["modality"][i]) # Sample(sample["path"][i], sample["modality"][i])
+            data = Sample(
+                None, sample["modality"]
+                [i])  # Sample(sample["path"][i], sample["modality"][i])
             # bboxes_3d = results[i]["box3d_lidar"].numpy()
```
Comment (LielinJiang) [inline paddle3d/models/detection/rtebev/rtebev.py]: remove this code

File: paddle3d/models/detection/rtebev/rtebev.py [hunk 6]
```diff
@@ -562,33 +575,34 @@ def _parse_results_to_sample(self, results: dict, sample: dict):
...
-                    # points,
-                    # img_metas,
-                    samples,
-                    img=None,
-                    rescale=False,
-                    **kwargs):
-        
+    def simple_test(
+            self,
+            # points,
```
Comment (LielinJiang) [inline paddle3d/models/detection/rtebev/rtebev.py]: remove this code

File: paddle3d/models/detection/rtebev/rtebev.py [hunk 7]
```diff
@@ -562,33 +575,34 @@ def _parse_results_to_sample(self, results: dict, sample: dict):
...
             points, img=img, img_metas=img_metas, **kwargs)
-        
+
         bbox_list = [dict() for _ in range(len(img_metas))]
         bbox_pts = self.simple_test_pts(img_feats, img_metas, rescale=rescale)
         for result_dict, pts_bbox in zip(bbox_list, bbox_pts):
             result_dict['pts_bbox'] = pts_bbox
-        
+
         #preds, x = self.pts_bbox_head(img_feats[0])
```
Comment (LielinJiang) [inline paddle3d/models/detection/rtebev/rtebev.py]: remove this code

File: paddle3d/models/detection/rtebev/rtebev.py [hunk 8]
```diff
@@ -611,12 +625,11 @@ def simple_test(self,
...
         # =================================================================
 
-
         return {"preds": [result_dict]}
 
     def simple_test_pts(self, x, img_metas, rescale=False):
         """Test function of point cloud branch."""
-        outs = self.pts_bbox_head(x) # single apply
+        outs = self.pts_bbox_head(x)  # single apply
         # outs = self.pts_bbox_head(x[0]) # single apply
```
Comment (LielinJiang) [inline paddle3d/models/detection/rtebev/rtebev.py]: remove this code

File: paddle3d/models/detection/rtebev/rtebev.py [hunk 9]
```diff
@@ -947,7 +943,9 @@ def shift_feature(self, input, trans, rots, bda, bda_adj=None):
         '''
 
         #l02l1 = l02l1[:, :, :, [True, True, False, True], :][:, :, :, :, [True, True, False, True]]
-        l02l1 = l02l1.index_select(paddle.to_tensor([0, 1, 3]), axis = 3).index_select(paddle.to_tensor([0, 1, 3]), axis = 4)
+        l02l1 = l02l1.index_select(
+            paddle.to_tensor([0, 1, 3]), axis=3).index_select(
+                paddle.to_tensor([0, 1, 3]), axis=4)
 
         feat2bev = paddle.zeros((3, 3), dtype=grid.dtype)
         #torch.zeros((3, 3), dtype=grid.dtype).to(grid)
```
Comment (LielinJiang) [inline paddle3d/models/detection/rtebev/rtebev.py]: remove

File: paddle3d/models/detection/rtebev/rtebev.py [hunk 10]
```diff
@@ -956,29 +954,33 @@ def shift_feature(self, input, trans, rots, bda, bda_adj=None):
         feat2bev[0, 2] = self.img_view_transformer.grid_lower_bound[0]
         feat2bev[1, 2] = self.img_view_transformer.grid_lower_bound[1]
         feat2bev[2, 2] = 1
-        feat2bev = feat2bev.reshape((1, 3, 3))#.view(1, 3, 3)
+        feat2bev = feat2bev.reshape((1, 3, 3))  #.view(1, 3, 3)
         tf = paddle.inverse(feat2bev).matmul(l02l1).matmul(feat2bev)
         #torch.inverse(feat2bev).matmul(l02l1).matmul(feat2bev)
```
Comment (LielinJiang) [inline paddle3d/models/detection/rtebev/rtebev.py]: remove

File: paddle3d/models/detection/rtebev/rtebev.py [hunk 11]
```diff
@@ -1263,7 +1273,7 @@ def forward_aux_pts_train(self,
         Returns:
             dict: Losses of each branch.
         """
-        outs = self.aux_pts_bbox_head(pts_feats[0]) # single apply
+        outs = self.aux_pts_bbox_head(pts_feats[0])  # single apply
         loss_inputs = [gt_bboxes_3d, gt_labels_3d, outs[0]]
         # outs = self.pts_bbox_head(pts_feats) # single apply
```
Comment (LielinJiang) [inline paddle3d/models/detection/rtebev/rtebev.py]: remove it

File: paddle3d/models/detection/rtebev/rtebev.py [hunk 12]
```diff
@@ -1302,19 +1312,20 @@ def forward(self, samples, *args, **kwargs):
...
-    def forward_test(self,
-                    samples,
-                    #  points=None,
-                    #  img_metas=None,
-                    #  img_inputs=None,
-                     **kwargs):
+    def forward_test(
+            self,
+            samples,
+            #  points=None,
```
Comment (LielinJiang) [inline paddle3d/models/detection/rtebev/rtebev.py]: remove it

File: paddle3d/models/detection/rtebev/rtebev.py [hunk 13]
```diff
@@ -1498,24 +1511,257 @@ def bbox3d2result(self, bboxes, scores, labels, attrs=None):
...
             points, img=img_inputs[0], img_metas=img_metas, **kwargs)
         #assert self.with_pts_bbox
-        outs = self.pts_bbox_head(img_feats[0])# single apply in paddle centerhead
+        outs = self.pts_bbox_head(
+            img_feats[0])  # single apply in paddle centerhead
         return outs
-    
 
     # ==========================
     # format torch style output result
```
Comment (LielinJiang) [inline paddle3d/models/detection/rtebev/rtebev.py]: remove it

File: paddle3d/models/detection/rtebev/rtebev.py [hunk 14]
```diff
@@ -1628,8 +1875,8 @@ def forward(self, x, pe=None):
         x = self.out_conv(x)
         return x
 
-class LightDepthReducer(nn.Layer):
 
+class LightDepthReducer(nn.Layer):
```
Comment (LielinJiang) [inline paddle3d/models/detection/rtebev/rtebev.py]: remove this class

File: paddle3d/models/transformers/bevdet_transformer.py [hunk 15]
```diff
@@ -519,21 +511,21 @@ def __init__(self,
         )
         self.context_conv = nn.Conv2D(
             mid_channels, context_channels, kernel_size=1, stride=1, padding=0)
-        self.bn = nn.BatchNorm1D(27) 
-        self.depth_mlp = Mlp(27, mid_channels, mid_channels) # ?
-        self.depth_se = SELayer(mid_channels)  # NOTE: add camera-aware #? 
-        self.context_mlp = Mlp(27, mid_channels, mid_channels) # ? 
+        self.bn = nn.BatchNorm1D(27)
+        self.depth_mlp = Mlp(27, mid_channels, mid_channels)  # ?
+        self.depth_se = SELayer(mid_channels)  # NOTE: add camera-aware #?
```
Comment (LielinJiang) [inline paddle3d/models/transformers/bevdet_transformer.py]: remove note

Comment (wangna11BD) [reply paddle3d/models/transformers/bevdet_transformer.py]: Done

File: paddle3d/models/detection/rtebev/rtebev.py (hunk 14 above)
Comment (wangna11BD) [reply paddle3d/models/detection/rtebev/rtebev.py]: Done

File: paddle3d/models/detection/rtebev/rtebev.py (hunk 13 above)
Comment (wangna11BD) [reply paddle3d/models/detection/rtebev/rtebev.py]: Done

File: paddle3d/models/detection/rtebev/rtebev.py (hunk 12 above)
Comment (wangna11BD) [reply paddle3d/models/detection/rtebev/rtebev.py]: Done

File: paddle3d/models/detection/rtebev/rtebev.py (hunk 11 above)
Comment (wangna11BD) [reply paddle3d/models/detection/rtebev/rtebev.py]: Done

File: paddle3d/models/detection/rtebev/rtebev.py (hunk 10 above)
Comment (wangna11BD) [reply paddle3d/models/detection/rtebev/rtebev.py]: Done

File: paddle3d/models/detection/rtebev/rtebev.py (hunk 9 above)
Comment (wangna11BD) [reply paddle3d/models/detection/rtebev/rtebev.py]: Done

File: paddle3d/models/detection/rtebev/rtebev.py (hunk 8 above)
Comment (wangna11BD) [reply paddle3d/models/detection/rtebev/rtebev.py]: Done

File: paddle3d/models/detection/rtebev/rtebev.py (hunk 7 above)
Comment (wangna11BD) [reply paddle3d/models/detection/rtebev/rtebev.py]: Done

File: paddle3d/models/detection/rtebev/rtebev.py (hunk 6 above)
Comment (wangna11BD) [reply paddle3d/models/detection/rtebev/rtebev.py]: Done

File: paddle3d/models/detection/rtebev/rtebev.py (hunk 5 above)
Comment (wangna11BD) [reply paddle3d/models/detection/rtebev/rtebev.py]: Done

File: docs/models/rtebev/README.md (hunk 1 above)
Comment (wangna11BD) [reply docs/models/rtebev/README.md]: Done

File: docs/models/rtebev/README.md (hunk 2 above)
Comment (wangna11BD) [reply docs/models/rtebev/README.md]: Done

File: docs/models/rtebev/README.md (hunk 3 above)
Comment (wangna11BD) [reply docs/models/rtebev/README.md]: Done

**Review by LielinJiang — APPROVED** [review]
